src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -64,13 +67,73 @@ impl ops::AddAssign for ItemCount {
     }
 }
 
-#[derive(Default)]
 struct CoverageCalculator {
     items: BTreeMap<FileName, ItemCount>,
+    output_format: Option<OutputFormat>,
+}
+
+fn limit_filename_len(filename: String) -> String {
+    let nb_chars = filename.chars().count();
+    if nb_chars > 35 {
+        "...".to_string()
+            + &filename[filename.char_indices().nth(nb_chars - 32).map(|x| x.0).unwrap_or(0)..]
+    } else {
+        filename
+    }
+}
+
+#[derive(Serialize)]
+struct CoverageCalculatorEntry {
```

Comment:
With the percentage removed this struct isn't needed, `#[derive(Serialize)]` can be added to `ItemCount` instead.